DPGAnalysis/CommonNanoAOD/plugins/HitPositionTableProducer.cc

```diff
@@ -0,0 +1,119 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/NanoAOD/interface/FlatTable.h"
+#include "DataFormats/Common/interface/View.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHitContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+#include "Geometry/Records/interface/GlobalTrackingGeometryRecord.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "DataFormats/CaloRecHit/interface/CaloRecHit.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+#include "RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h"
+
+#include <vector>
+#include <iostream>
+
+template <typename T>
+class HitPositionTableProducer : public edm::stream::EDProducer<> {
+public:
+  HitPositionTableProducer(edm::ParameterSet const& params)
+      : name_(params.getParameter<std::string>("name")),
+        doc_(params.getParameter<std::string>("doc")),
+        src_(consumes<T>(params.getParameter<edm::InputTag>("src"))),
+        cut_(params.getParameter<std::string>("cut"), true) {
+    produces<nanoaod::FlatTable>();
+  }
+
+  ~HitPositionTableProducer() override {}
+
+  void beginRun(const edm::Run&, const edm::EventSetup& iSetup) override {
+    // TODO: check that the geometry exists
+    iSetup.get<CaloGeometryRecord>().get(caloGeom_);
+    rhtools_.setGeometry(*caloGeom_);
+    iSetup.get<TrackerDigiGeometryRecord>().get("idealForDigi", trackGeom_);
+    // Believe this is ideal, but we're not so precise here...
+    iSetup.get<GlobalTrackingGeometryRecord>().get(globalGeom_);
+  }
```

Comment:
ES consumes should be used

based on issues reported in the static analysis
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-86974f/13067/llvm-analysis/